dashboard/test/controllers/api/v1/pd/applications_controller_test.rb

```diff
@@ -403,6 +403,66 @@ class ApplicationsControllerTest < ::ActionController::TestCase
       assert_equal expected_log, @csp_facilitator_application.sanitize_status_timestamp_change_log
     end
 
+    test 'update appends to timestamp log if workshop admin changes application from unlocked to locked' do
+      sign_in @workshop_admin
+      @csp_facilitator_application.update(status_timestamp_change_log: '[]', locked_at: nil)
+      @csp_facilitator_application.reload
+
+      assert_equal [], @csp_facilitator_application.sanitize_status_timestamp_change_log
+      assert_equal false, @csp_facilitator_application.locked?
+
+      # Changing application from unlocked to locked
+      post :update, params: {id: @csp_facilitator_application.id, application: {status: @csp_facilitator_application.status, locked: true}}
+      @csp_facilitator_application.reload
+
+      expected_log = [
+        {
+          title: 'Application is locked',
+          changing_user_id: @workshop_admin.id,
+          changing_user_name: @workshop_admin.name,
+          time: Time.zone.now
+        }
+      ]
+
+      assert_equal expected_log, @csp_facilitator_application.sanitize_status_timestamp_change_log
+
+      # Setting application to locked again
+      post :update, params: {id: @csp_facilitator_application.id, application: {status: @csp_facilitator_application.status, locked: true}}
+      @csp_facilitator_application.reload
+
+      assert_equal expected_log, @csp_facilitator_application.sanitize_status_timestamp_change_log
+    end
+
+    test 'update appends to timestamp log if workshop admin changes application from locked to unlocked' do
+      sign_in @workshop_admin
+      @csp_facilitator_application.update(status_timestamp_change_log: '[]', locked_at: Time.zone.now)
+      @csp_facilitator_application.reload
+
+      assert_equal [], @csp_facilitator_application.sanitize_status_timestamp_change_log
```

Comment:
It's not necessary to test this behavior again

Reply:
Actually this part is to prepare the correct starting condition for the test. We want to make sure the application is locked first.